defaultdict.default_factory is not updatable.

One of the interesting things about `defaultdict.default_factory` is that you can set it to `None` to make your defaultdict behave like a normal dict (e.g. start raising `KeyError` when things aren't found.  However, if you try this with `mypy --strict`, you'll have problems:

```py
# ~/sandbox/dd.py
from typing import DefaultDict, Mapping
from collections import defaultdict

def raz_a_ma_taz() -> Mapping[int, int]:
    result: DefaultDict[int, int] = defaultdict(int)
    for i in range(10):
        result[i] += 6
    result.default_factory = None
    return result
```

```sh
$ mypy --strict ~/sandbox/dd.py 
~/sandbox/dd.py:8: error: Cannot assign to a method
~/sandbox/dd.py:8: error: Attribute function "default_factory" with type "Callable[[], int]" does not accept self argument
~/sandbox/dd.py:8: error: Incompatible types in assignment (expression has type "None", variable has type "Callable[[], int]")
```

I'm running mypy 0.620 in all of my anaconda environments and I've tried this in my python 3.6.1 and python 3.7 environments.

I have to admit that I'm not completely sure where this error is coming from. 
 https://github.com/python/typeshed/blob/master/stdlib/3/collections/__init__.pyi#L307 looks promising (`default_factory` should probably be `Optional` there),  but it still doesn't explain why mypy thinks that it's a method rather than an attribute that happens to be callable.  I'm also not sure that I understand how the stub for `DefaultDict` from [`typing.pyi`](https://github.com/python/typeshed/blob/master/stdlib/3/typing.pyi#L45) works (or if that is at all related).

--- a/stdlib/3/collections/__init__.pyi
+++ b/stdlib/3/collections/__init__.pyi
@@ -305,7 +305,7 @@ class OrderedDict(Dict[_KT, _VT], Reversible[_KT], Generic[_KT, _VT]):
 _DefaultDictT = TypeVar('_DefaultDictT', bound=defaultdict)
 
 class defaultdict(Dict[_KT, _VT], Generic[_KT, _VT]):
-    default_factory = ...  # type: Callable[[], _VT]
+    default_factory = ...  # type: Optional[Callable[[], _VT]]
 
     @overload
     def __init__(self, **kwargs: _VT) -> None: ...

